packages/zoltan2/core/src/algorithms/color/Zoltan2_AlgHybrid2GL.hpp

```diff
@@ -0,0 +1,1532 @@
+      Kokkos::View<int*, device_type> verts_to_recolor_size("verts to recolor size",1);
+      Kokkos::View<int*, device_type, Kokkos::MemoryTraits<Kokkos::Atomic>> verts_to_recolor_size_atomic = verts_to_recolor_size;
+      typename Kokkos::View<int*, device_type>::HostMirror verts_to_recolor_size_host = create_mirror(verts_to_recolor_size);
+      
+      //initialize the host view
+      verts_to_recolor_size_host(0) = 0;
+      //copy to device
+      Kokkos::deep_copy(verts_to_recolor_size, verts_to_recolor_size_host);
+
+      //verts to send can only include locally owned vertices,
+      //so we can safely allocate the view to size n_local.
+      //
+      //This view is a list of local vertices that are going to be
+      //recolored, and need to be sent to their remote copies.
+      Kokkos::View<lno_t*, device_type> verts_to_send_view("verts to send", n_local);
+      Kokkos::View<lno_t*, device_type, Kokkos::MemoryTraits<Kokkos::Atomic>> verts_to_send_atomic = verts_to_send_view;
+      typename Kokkos::View<lno_t*, device_type>::HostMirror verts_to_send_host = create_mirror(verts_to_send_view);
+      
+      //this view keeps track of the size of verts_to_send.
+      Kokkos::View<size_t*, device_type> verts_to_send_size("verts to send size",1);
+      Kokkos::View<size_t*, device_type, Kokkos::MemoryTraits<Kokkos::Atomic>> verts_to_send_size_atomic = verts_to_send_size;
+      typename Kokkos::View<size_t*, device_type>::HostMirror verts_to_send_size_host = create_mirror(verts_to_send_size);
+
+      verts_to_send_size_host(0) = 0;
+      Kokkos::deep_copy(verts_to_send_size, verts_to_send_size_host);
+
+      if(verbose) std::cout<<comm->getRank()<<": Constructing the boundary\n";
+      
+      //this function allocates and initializes the boundary_verts_dev view,
+      //sets the boundary_size accordingly, and adds vertices to the
+      //verts_to_send_atomic view, updating the size view as well.
+      constructBoundary(n_local, dist_offsets_dev, dist_adjs_dev, dist_offsets_host, dist_adjs_host, boundary_verts_dev, 
+	                boundary_size, verts_to_send_atomic, verts_to_send_size_atomic);
+      
+      //this boolean chooses which KokkosKernels algorithm to use.
+      //It was experimentally chosen for distance-1 coloring.
+      //Should be disregarded for distance-2 colorings.
+      bool use_vbbit = (global_max_degree < 6000);
+      //Done initializing, start coloring!
+      comm->barrier();
+      interior_time = timer();
+      total_time = timer();
+      //give the entire local graph to KokkosKernels to color
+      this->colorInterior(n_local, adjs_dev, offsets_dev, femv,adjs_dev,0,use_vbbit);
+      interior_time = timer() - interior_time;
+      comp_time = interior_time;
+      //get the color view from the FEMultiVector
+      Kokkos::View<int**, Kokkos::LayoutLeft> femvColors = femv->template getLocalView<memory_space>();
```

Comment:
Prefer use of auto for type.  Also, if getting a device view, use femv->getLocalViewDevice().